src/custom/state/claim/hooks/index.ts

```diff
@@ -288,17 +322,26 @@ function useDeploymentTimestamp(): number | null {
   const { chainId } = useActiveWeb3React()
   const vCowContract = useVCowContract()
   const [timestamp, setTimestamp] = useState<number | null>(null)
+  const isMounted = useIsMounted()
 
   useEffect(() => {
     if (!chainId || !vCowContract) {
       return
     }
 
-    vCowContract.deploymentTimestamp().then((ts) => {
-      console.log(`Deployment timestamp in seconds: ${ts.toString()}`)
-      setTimestamp(ts.mul('1000').toNumber())
-    })
-  }, [chainId, vCowContract])
+    fetchDeploymentTimestamp(vCowContract)
+      .then((timestamp) => {
+        if (isMounted.current) {
+          setTimestamp(timestamp)
+        }
+      })
+      .catch(() => {
+        if (isMounted.current) {
+          setTimestamp(timestamp)
```

Comment:
Where are you getting the timestamp from here when it fails?

I think you don't want to set anything here.

Also, you should probably not depend on outer scope `timestamp` here because on every update you'll try to update it again.

Removing this line should fix it, though

Reply:
My mistake, it should be null there

Reply:
I didn't test for that scenario that it fails but just in case it does I think that should cover it